src/System.Private.CoreLib/shared/System/Threading/Thread.cs

```diff
@@ -17,6 +17,9 @@ public sealed partial class Thread : CriticalFinalizerObject
     {
         private static AsyncLocal<IPrincipal> s_asyncLocalPrincipal;
```

Comment:
You can move  `_executionContext` to shared too.

Reply:
done